x-pack/plugins/transform/public/app/sections/transform_management/components/action_delete/delete_action_name.tsx

```diff
@@ -24,45 +26,71 @@ export const deleteActionNameText = i18n.translate(
 );
 
 const transformCanNotBeDeleted = (i: TransformListRow) =>
+  i.stats &&
   !([TRANSFORM_STATE.STOPPED, TRANSFORM_STATE.FAILED] as TransformState[]).includes(i.stats.state);
 
 export const isDeleteActionDisabled = (items: TransformListRow[], forceDisable: boolean) => {
   const disabled = items.some(transformCanNotBeDeleted);
-  return forceDisable === true || disabled;
+  const hasNoStats = items.some((i: TransformListRow) => !isTransformListRowWithStats(i));
```

Comment:
Nit: This same line to get `hasNoStats` we have now multiple times across the code, maybe it's worth moving to a helper function.

Reply:
Updated here [`26be102` (#166320)](https://github.com/elastic/kibana/pull/166320/commits/26be102655aa013f2fc026a0c6bc8e3f900218c9)